crates/accelerate/src/sampled_exp_val.rs

```diff
@@ -83,12 +116,41 @@ pub fn sampled_expval_complex(
         .into_iter()
         .enumerate()
         .map(|(idx, string)| coeff_arr[idx] * c64(bitstring_expval(&dist, string), 0.))
-        .sum();
+        .sum::<Complex64>();
     Ok(out.re)
 }
 
+// Compute the expectation value from a sampled distribution for SparseObservable objects
+#[pyfunction]
+#[pyo3(text_signature = "(sparse_obs, dist, /)")]
+pub fn sampled_expval_sparse_observable(
+    sparse_obs: PyRef<PySparseObservable>,
+    dist: HashMap<String, f64>,
+) -> PyResult<f64> {
+    // Access the SparseObservable
+    let sparse_obs: SparseObservable = sparse_obs.get().unwrap();
+    // Convert SparseObservable to operator strings and coefficients
+    let result: Result<Complex64, PyErr> =
+        sparse_obs
+            .iter()
+            .enumerate()
+            .try_fold(Complex64::new(0.0, 0.0), |acc, (_idx, term)| {
+                if ![BitTerm::Z, BitTerm::Zero, BitTerm::One]
+                    .contains(term.bit_terms.iter().next().unwrap())
```

Comment:
Doesn't this just check whether the first bit term in the `term` is diagonal? What if `term.bit_terms` is something like `[Z, X]`? To avoid iterating over the bit terms twice, we could delegate this check to `bitstring_expval_bitterm` and maybe replace the `fold` with a `try_fold` to handle this check and return an error if we encounter non-diagonal terms